docs/DeveloperGuide.md

```diff
@@ -4,35 +4,261 @@
+
+![Parser Class Diagram](images/class/Parser.png)
+
+*Figure 4: Parser Class Diagram*
+
+Note that `CafeCtrl` only have access to the interface `ParserUtil` although the run-time type object is `Parser`. With this, we are able to decrease coupling between `CafeCtrl` and `Parser`, allowing for easier maintenance. This also ensures the testability as we could provide mock or stub dependencies during testing, we could isolate the behavior of the class and focus on unit testing without external dependencies. 
+
+![Parser Parsing User Input Sequence Diagram](images/sequence/Parser.png)
+
+*Figure 5: Parser Parsing User Input Sequence Diagram*
+
+When user input a string to `Main`,  it passes the full user input to `Parser` via `parseCommand`. In `parseCommand`,  it finds the matching keyword for different command from the user input, then it calls the respective `prepareCommand` method within `Parser`. `prepareCommand` then generates the corresponding command class and return it to `parseCommand`, which returns the `Command` back to `Main` for execution.
+
+### Storage Component
+API: [Storage.java]({repoURL}src/main/java/seedu/cafectrl/storage/Storage.java)
+
+![Storage Class Diagram](images/class/Storage.png)
+
+*Figure 6: Storage Class Diagram*
+
+The `Storage` class,
+- loads and saves the list of dishes on the `Menu`, available ingredient stock in `Pantry` and orders for the day in `OrderList` in a text file.
+- depends on `Menu`, `Pantry` and `Sales` objects (which are found in the data package).
+- is composed of `FileManager` object as the text file needs to be located first before reading or writing.
+
+### Data Component
+API: []
+![Data Class Diagram](images/class/Data.png)
+*Figure 7: Data Package Class Diagram*
+
+The 'Data' package consists of all the classes that the commands interact with to perform various functions.
+A summary of the class diagram is as listed below:
+- Each `Dish` within the `Menu` is constructed with a set of `Ingredient` instances, forming a one-to-many relationship with `Ingredient`.
+- `Pantry` is instantiated with an ArrayList of `Ingredients` (`pantryStock`), forming a one-to-many relationship with `Ingredient`.
+- The `Chef` class has a one-to-one relationship with `Pantry`, ensuring access to necessary ingredients for dish preparation.
+- When an order is placed, the `Order` class is instantiated with an ArrayList of `Ingredient` (`ingredientList`), forming a one-to-many relationship with `Ingredient`.
+- `OrderList` is instantiated with an ArrayList of `Order`, forming a one-to-many relationship with `Order`.
+- `Sales` is instantiated with an ArrayList of `OrderList`, forming a one-to-many relationship with `OrderList`.
+- Lastly, the `CurrentDate` class keeps track of the current operating day of the cafe.
+
+
+## Features
+
+### Add Dish
+
+![Add Dish Execution](images/sequence/AddDishCommand_execute.png)
+*Figure 8: Execution of add_dish command*
```

Comment:
The "Figure 8" text is in a weird location besides the diagram
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/49584216/37d269c4-d8bd-4159-8cbe-041e3de30d7c)

Reply:
oops I forgot to add a newline thanks for pointing it out 👍